### ISSUE TITLE
Lumen 5.4 - 5.6: Call to undefined method Laravel\Lumen\Application::routesAreCached()

### ISSUE THREAD

**Issue by GTCrais** [laravel/lumen-framework#746]
- Lumen Version: 5.4 - 5.6
- PHP Version: 7.1.10

### Description (and steps to reproduce):
You can call `$this->loadRoutesFrom()` within a `ServiceProvider`, which then calls `$this->app->routesAreCached()`, but that method does not exist.

**Comment by dunhamjared** [comment]
Verified issue in v5.6.3

**Comment by dunhamjared** [comment]
A quick dirty hack is to overwrite the function it's trying to call in the service provider. 

So in your `class AppServiceProviderExample extends ServiceProvider` class, add:

```php
//@TODO Remove me when a fix is implemented
public function loadRoutesFrom($location){
    return $location;
}
````

It's trying to call the `routesAreCached` function from `Illuminate\Foundation\Application` but `$this->app` is in `Laravel\Lumen\Application`.

**Comment by 4levels** [comment]
Hi Lumen team,

Any updates on this?  I'm also facing the exact same issue and all be it that the proposed fix works, this doesn't work for libraries that don't extend the AppServiceProvider (but use the Lumen/Support/ServiceProvider directly).

Kind regards,

**Comment by JacksonYara** [comment]
I don't think route caching is supported in Lumen. Maybe, this could be solved by adding the `routesAreCached` method to the `Laravel\Lumen\Application`, which returns false.

```PHP
/**
 * Determine if the application routes are cached.
 *
 * @return bool
 */
public function routesAreCached()
{
    return false;
}
```

**Comment by driesvints** [comment]
This is fixed in Lumen 5.7
